scripts/control-plane-dpu/unit-tests/test_bfb_discovery.sh

```
@@ -0,0 +1,80 @@
+
+assert_eq "single BFB found" "bf-bundle-3.2.2.bfb" "$(
+    with_tmpdir bash -c '
+        touch bf-bundle-3.2.2.bfb
+        discover_bfb() {
+            local -a _bfbs=(bf-bundle*.bfb)
+            [ ! -f "${_bfbs[0]}" ] && { echo "ERROR: not found" >&2; return 1; }
+            [ "${#_bfbs[@]}" -gt 1 ] && { echo "ERROR: multiple: ${_bfbs[*]}" >&2; return 1; }
+            echo "${_bfbs[0]}"
+        }
+        discover_bfb
+    '
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)